TorchTensor/NativeMemoryManager.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System.Buffers;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System;
+
+namespace TorchTensor
+{
+    public class NativeMemory<T> : MemoryManager<T>
```

Comment:
Can this class be `internal`?

Reply:
I would like to have it internal, but in that way I can create mockup classes in the unit tests